lepton/starterkits/model_compression.ipynb

```diff
@@ -0,0 +1,701 @@
+    "\n",
+    "**NOTE:** The default configuration in the notebook runs on 8 x 80GB NVIDIA GPUs. However, you can potentially reduce the Tensor Parallel size (`TENSOR_PARALLEL_SIZE`) along with the Micro-Batchsize (`MICRO_BATCH_SIZE`) in the distillation scripts to accommodate lower resource availability.\n",
+    "\n",
+    "## Create a Pruned and Distilled Model with NeMo Framework\n",
+    "\n",
+    "For pruning and distilling the model, you will use the NeMo Framework, which is available as a [Docker container](https://catalog.ngc.nvidia.com/orgs/nvidia/containers/nemo). These notebooks has been tested on `nvcr.io/nvidia/nemo:25.09` container with `nvidia-modelopt==0.35.1`.\n",
+    "\n",
+    "1. Run the container using the following command. You will mount your local directory to `/workspace` so the model and dataset will be stored in a persistent location. If you are using your own model and dataset, you can change the paths in the notebooks accordingly.\n",
+    "\n",
+    "```bash\n",
+    "export FW_VERSION=25.09\n",
+    "```\n",
+    "\n",
+    "```bash\n",
+    "docker run \\\n",
+    "  --gpus all \\\n",
+    "  --shm-size=16GB \\\n",
+    "  --net=host \\\n",
+    "  --ulimit memlock=-1 \\\n",
+    "  --rm -it \\\n",
+    "  -v ${PWD}:/workspace \\\n",
+    "  -w /workspace \\\n",
+    "  nvcr.io/nvidia/nemo:$FW_VERSION bash\n",
+    "```\n",
+    "\n",
+    "2. From within the container, copy the notebooks to your local directory so changes remain persistent (only if running first time).\n",
+    "\n",
+    "```bash\n",
+    "cp -r /opt/NeMo/tutorials/llm/qwen/pruning-distillation/* /workspace\n",
+    "```\n",
+    "\n",
+    "3. From within the container, login with your Hugging Face token to download any gated models or datasets (not required if you have already downloaded the model and datasets).\n",
+    "\n",
+    "```bash\n",
+    "huggingface-cli login --token <YOUR_HF_ACCESS_TOKEN>\n",
+    "```\n",
+    "\n",
+    "4. Start the Jupyter lab:\n",
+    "\n",
+    "```bash\n",
+    "pip install --upgrade ipywidgets notebook\n",
+    "jupyter lab --ip 0.0.0.0 --port=8888 --allow-root\n",
+    "```\n",
+    "\n",
+    "5. Then, navigate to this directory which contains a list of notebooks that cover all the steps to create a distilled 6B model from Qwen3-8B.\n",
+    "\n",
+    "This workflow is structured into four notebooks:\n",
+    "  1. [Prepare the model and dataset](./01_model_and_data_preparation.ipynb) to convert HuggingFace model to NeMo format and tokenize the dataset\n",
+    "  2. [Prune the model](./02_pruning.ipynb) to create a pruned model via either depth-pruning or width-pruning\n",
+    "  3. [Distill knowledge from teacher into pruned student](./03_distillation.ipynb)\n",
+    "  4. [Compare depth and width pruned models](./04_depth_vs_width_pruning_comparison.ipynb)\n"
```

Comment:
Since this is one notebook now, I'd just point to the four high-level sections but all in this notebook.